src/perturb.jl

```diff
@@ -40,57 +38,43 @@ function create_perturbation_gates(s, k, B_pert, α, x, L, N_sites, energy_sign,
             s_i = s[i]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
             s_j = s[j]
             # assert B vector to have a magnitude of 1 while preserving its direction.
-            @assert norm(B) == 1
+            @assert norm(B_pert) == 1
             # total Hamiltonian of the system is a sum of local terms hj, where hj acts on sites i and j which are paired for gates to latch onto.
             # op function returns these operators as ITensors and we tensor product and add them together to compute the operator hj.
 
             # add perturbation via one-body oscillation term to the Hamiltonian
-            # hj = (1/(N_sites-1))* (
-            # ((ω_pert[i] * B[1] * op("Sx", s_i)* op("Id", s_j))  + (ω_pert[j] * B[1] * op("Sx", s_j) * op("Id", s_i))) + 
-            # ((ω_pert[i] * B[2] * op("Sy", s_i)* op("Id", s_j))  + (ω_pert[j] * B[2] * op("Sy", s_j) * op("Id", s_i))) +
-            # ((ω_pert[i] * B[3] * op("Sz", s_i)* op("Id", s_j))  + (ω_pert[j] * B[3] * op("Sz", s_j) * op("Id", s_i))) 
-            # )
-            hj= (1/(N_sites-1))* (
+            hj= (
```

Comment:
Why is this looping over sites twice and using a two-body operator? Shouldn't we just loop over sites once and apply the perturbation to each site separately? Does this mean that the perturbation will be larger if the number of sites is larger (since it gets applied multiple times)?